Database/MySQLMPOS.py

```diff
@@ -18,46 +18,55 @@ def get_uid(self, username):
 	    try:
 		with connection.cursor() as cursor:
 		   user = username.split('.', 1)[0]
-		   cursor.execute("SELECT id FROM accounts where username = %s", (uname,))
+		   cursor.execute("SELECT `id`, `is_locked` FROM `accounts` where `username`='{0}'".format(user))
         	   row = cursor.fetchone()
 		   if row is None:
 			return False
+		   elif	row['is_locked'] == 2:
+			self.logger.error("User {} is banned by admin".format(user))
+			return False
 		   else:
 			return row['id']
 		   return False
-	    except:
-		self.logger.error(e)
+	    except Exception :
 		return False
 
 	def authorise(self, *args, **kwargs):
 	    self.logger.info("Checking Auth For User: {0}".format(kwargs['username']))
 	    connection = self.connect()
-	    try:
+	    uid = self.get_uid(kwargs['username'])
+	    if uid:
+	     try:
 		with connection.cursor() as cursor:
-		   cursor.execute("SELECT id FROM `pool_worker` WHERE `username` = %(username)s", kwargs)
+		   cursor.execute("SELECT `id` FROM `pool_worker` WHERE `username` = %(username)s", kwargs)
 		   result = cursor.fetchone()
-		   if result is None and not kwarg['create_user']:
-			return 0
+		   self.logger.info(kwargs)
+		   if result is None and not kwargs['create_user']:
+			return 0 
 		   elif result is None and kwargs['create_user']:
-			uid = self.get_uid(kwargs['username'])
-			if uid:
-        		   query = "INSERT INTO pool_worker (account_id, username, password) VALUES (%s, %s, 'x');", (uid, kwargs['username'])
-		           cursor.execute(query)
-			   connection.commit()
+			   self.logger.info("User  {} authorized, but have no workers.".format(kwargs['username']))
 			   return self.authorise()
 		   return result['id']
-	    except Exception as e:
+	     except Exception as e:
 		self.logger.error(e)
 		return False
-	    return False
+	     return False
 
         def insert_shares(self, *args, **kwargs):
 	    kwargs['lres'] = 'Y' if kwargs['lres'] == True else 'N'
             connection = self.connect()
             try:
                 with connection.cursor() as cursor:
-            	   cursor.execute("""INSERT INTO `shares` (time, rem_host, username, our_result, upstream_result, reason, solution, difficulty)
+                   cursor.execute("SELECT `id` FROM `pool_worker` WHERE `username` = %(uname)s", kwargs)
+                   result = cursor.fetchone()
+                   if result is None    :
+			uid = self.get_uid(kwargs['uname'])
+                        query = "INSERT INTO pool_worker (`account_id`, `username`, `password`) VALUES ({}, '{}', 'x');".format(uid, kwargs['uname'])
```

Comment:
Dont hardcode the worker's password. Some people have custom difficulty and other password pased parameters

Reply:
it was your hardcode)))